### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by KareemWilson**
### **In this Branch, I did the following:** 

1. Add a column species of type string to my animals' table
Add a column species of type string

2. Insert more animals.
insert more data

3. Inside a transaction, update the animals' table by setting the species column to unspecified
update the animals table by setting the species column to unspecified

4. Then roll back the change.
ROLLBACK

5. **Inside a transaction:**
- Update the animals' table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to digimon

- Update the animals' table by setting the species column to pokemon for all animals that don't have species already set.

- Commit the transaction.
Update the animals table by setting the species column to pokemon

6. **Inside a transaction** delete all records in the animals' table.
7. Then rollback the transaction.
Delete all records and rollback

8. **Inside a transaction:**
- Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
delete some animals and rollback to savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction
update weight of animals   0

9. Write  some queries to answer the following questions:

- How many animals are there?
How many animals are there?

- How many animals have never tried to escape?
How many animals have never tried to escape?

- What is the average weight of animals?
What is the average weight of animals?

- Who escapes the most, neutered or not neutered animals?
Who escapes the most, neutered or not neutered animals?

- What is the minimum and maximum weight of each type of animal?
min and max weight_kg

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
average between 1990 and 2000

### CODE REVIEW

**Review by omarsalem7 — COMMENTED** [review]
Hi @KareemWilson , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 💀 

You have submitted a project, that doesn't have an updated readme version, kindly update your readme with this [new one](https://github.com/microverseinc/readme-template) 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: queries.sql [hunk 1]
```diff
@@ -5,4 +5,63 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon','Pikachu');
...
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+/* Begin Transaction */
+BEGIN TRANSACTION;
+/* UPDATE species columns to 'unspecified' */
+UPDATE animals SET species = "unspecified";
+/* roll back the change */ 
+ROLLBACK;
```
Comment (EricMbouwe) [inline queries.sql]: - [x] Your transaction is missing a query to verify the change. Let's fix it
Hint: `SELECT species from animals; -- verify that change was made`

**Comment by KareemWilson** [comment]
Hi, @omarsalem7 
I've changed the required changes.

File: queries.sql [hunk 2]
```diff
@@ -5,4 +5,63 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon','Pikachu');
...
+
+/* Begin Transaction */
+BEGIN;
+/*Update the animals table by setting the species column to digimon for all animals that have a name ending in mon */
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+/*Update the animals table by setting the species column to pokemon for all animals that don't have species already set.*/
+UPDATE animals SET species = 'pokemon' WHERE species = '';
+/* COMMIT TRANSACTION */
+commit;
```
Comment (EricMbouwe) [inline queries.sql]: - [x] As mentioned above, let's add the verification query

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @KareemWilson, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- Create or change the structure of the tables ✔
- Populate tables ✔
- Query that helps to analyze data ✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Kareem, 
## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**